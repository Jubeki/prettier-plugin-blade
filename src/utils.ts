import { format } from "prettier";
// @ts-ignore
import php from "@prettier/plugin-php/standalone";

export const formatAsPhp = (source: string): string => {
  if (!source.startsWith("<?php")) {
    source = "<?php " + source;
  }

  let code = format(source, { parser: "php", plugins: [php] })
    .replace("<?php ", "")
    .trim();

  if (source.trim().endsWith(";")) {
    return code;
  }

  // The PHP plugin for Prettier will add a semi-colon by default. We don't always want that.
  return code.substring(0, code.length - 1);
};
