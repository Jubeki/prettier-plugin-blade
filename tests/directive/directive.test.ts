import { formatFile } from "../utils";
import path from "path";

it("escapes directives with parameters", () => {
  // Can still remove overflow of spaces, as multiple spaces will
  // be removed when rendered.
  const formatted = formatFile(
    path.join(__dirname, "./escaped-directive-parameters.blade.php")
  );
  expect(formatted).toMatchSnapshot();
});

it("escapes directives", () => {
  // Can still remove overflow of spaces, as multiple spaces will
  // be removed when rendered.
  const formatted = formatFile(
    path.join(__dirname, "./escaped-directive.blade.php")
  );
  expect(formatted).toMatchSnapshot();
});