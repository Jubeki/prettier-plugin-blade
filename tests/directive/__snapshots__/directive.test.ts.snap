// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escapes directives 1`] = `
"<h1>@@csrf</h1>
"
`;

exports[`escapes directives with parameters 1`] = `
"<h1>@@if( true )</h1>
"
`;
